modules/fiftyonedegrees/devicedetection/hook_raw_auction_request.go

```diff
@@ -0,0 +1,176 @@
+package devicedetection
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/golang/glog"
+	"github.com/prebid/prebid-server/v2/hooks/hookexecution"
+	"github.com/prebid/prebid-server/v2/hooks/hookstage"
+	"github.com/tidwall/gjson"
+	"github.com/tidwall/sjson"
+)
+
+func handleAuctionRequestHook(ctx hookstage.ModuleInvocationContext, deviceDetector deviceDetector, evidenceExtractor evidenceExtractor) (hookstage.HookResult[hookstage.RawAuctionRequestPayload], error) {
+	var result hookstage.HookResult[hookstage.RawAuctionRequestPayload]
+
+	// If the entrypoint hook was not configured, return the result without any changes
+	if ctx.ModuleContext == nil {
+		return result, hookexecution.NewFailure("entrypoint hook was not configured")
+	}
+
+	result.ChangeSet.AddMutation(
+		func(rawPayload hookstage.RawAuctionRequestPayload) (hookstage.RawAuctionRequestPayload, error) {
+			evidence, ua, err := evidenceExtractor.extract(ctx.ModuleContext)
+			if err != nil {
+				return rawPayload, hookexecution.NewFailure("error extracting evidence %s", err)
+			}
+			if evidence == nil {
+				return rawPayload, hookexecution.NewFailure("error extracting evidence")
+			}
+
+			deviceInfo, err := deviceDetector.getDeviceInfo(evidence, ua)
+			if err != nil {
+				return rawPayload, hookexecution.NewFailure("error getting device info %s", err)
+			}
+
+			result, err := hydrateFields(deviceInfo, rawPayload)
+			if err != nil {
+				er := fmt.Sprintf("error hydrating fields %s", err)
+				glog.Error(er)
+				return rawPayload, hookexecution.NewFailure(er)
```

Comment:
that's a leftover from the previous version of code, before your first review.
it had only the log part: `glog.Errorf("error hydrating fields %s", err)` but you suggested returning a hookexecution Failure error here instead of just logging. I'll remove the log here as it is redundant now.